src/adapters/erc721_collateralized_simple_interest/loan_adapter.ts

```diff
@@ -0,0 +1,698 @@
+     *
+     * @param {string} agreementId
+     * @param {TxData} options
+     * @returns {Promise<string>} The transaction's hash.
+     */
+    public async seizeCollateralAsync(agreementId: string, options?: TxData): Promise<string> {
+        this.assert.schema.bytes32("agreementId", agreementId);
+
+        const defaultOptions = await this.getTxDefaultOptions();
+
+        const transactionOptions = _.assign(defaultOptions, options);
+
+        await this.assertDebtAgreementExists(agreementId);
+        await this.assertCollateralSeizeable(agreementId);
+
+        const collateralizerContract = await this.contractsAPI.loadERC721CollateralizerAsync();
+
+        return collateralizerContract.seizeCollateral.sendTransactionAsync(
+            agreementId,
+            transactionOptions,
+        );
+    }
+
+    /**
+     * Returns collateral to the debt agreement's original collateralizer
+     * if and only if the debt agreement's term has lapsed and
+     * the total expected repayment value has been repaid.
+     *
+     * @param {string} agreementId
+     * @param {TxData} options
+     * @returns {Promise<string>} The transaction's hash.
+     */
+    public async returnCollateralAsync(agreementId: string, options?: TxData): Promise<string> {
+        this.assert.schema.bytes32("agreementId", agreementId);
+
+        await this.assertDebtAgreementExists(agreementId);
+        await this.assertCollateralReturnable(agreementId);
+
+        const defaultOptions = await this.getTxDefaultOptions();
+
+        const transactionOptions = _.assign(defaultOptions, options);
+
+        const collateralizerContract = await this.contractsAPI.loadERC721CollateralizerAsync();
+
+        return collateralizerContract.returnCollateral.sendTransactionAsync(
+            agreementId,
+            transactionOptions,
+        );
+    }
+
+    public unpackParameters(
+        termsContractParameters: string,
+    ): ERC721CollateralizedSimpleInterestTermsContractParameters {
+        const simpleInterestParams = this.simpleInterestLoanTerms.unpackParameters(
+            termsContractParameters,
+        );
+
+        const collateralizedParams = this.collateralizedLoanTerms.unpackParameters(
+            termsContractParameters,
+        );
+
+        return {
+            ...simpleInterestParams,
+            ...collateralizedParams,
+        };
+    }
+
+    public packParameters(
+        simpleTermsParams: SimpleInterestTermsContractParameters,
+        collateralTermsParams: ERC721CollateralizedTermsContractParameters,
+    ): string {
+        const packedSimpleInterestParams = this.simpleInterestLoanTerms.packParameters(
+            simpleTermsParams,
+        );
+
+        const packedCollateralizedParams = this.collateralizedLoanTerms.packParameters(
+            collateralTermsParams,
+        );
+
+        // Our final output is the perfect union of the packed simple interest params and the packed
+        // collateralized params.
```

Comment:
I think this comment is slightly misleading, since you're actually replacing some of the packed simple interest params with the packed ERC-721 collateralized params